util.go

```diff
@@ -11,3 +16,47 @@ var KValue = 20
 // Deprecated: This value is still the default, but changing this parameter no longer does anything.
 // Alpha is the concurrency factor for asynchronous requests.
 var AlphaValue = 3
+
+// RoutabilityBasedModeSwitching uses LocalRoutabilty events sent over the DHT's EventBus to trigger switching between
+// client and server modes.
+func RoutabilityBasedModeSwitching(dht *IpfsDHT) {
```

Comment:
this feels like a bit of an awkward file to put this in, if anyone feels strongly we can move it to another file or to a new package.

Reply:
@Stebalien doing things this way eats an extra goroutine compared to what it could be if it was embedded in the DHT. Is it worth trying to improve the API here to not need the extra goroutine or is it cool to just rework the API later if we care?

Reply:
Personally, I don't have a strong preference. One single one-off goroutine isn't going to matter in the long run. I'd say leave it here as it decouples the code and then move it later if someone objects.